docs/source/cpp_api/core_math.rst

```diff
@@ -0,0 +1,18 @@
+Mathematical Functions
+======================
+
+.. role:: py(code)
+   :language: c++
+   :class: highlight
+
+
+The math functions APIs guarantee both CUDA and CPU compatibility, making it more straightforward to write `__host__ __device__` functions without being concerned whether the underlying intrinsics will build and work.
```

Comment:
```suggestion
The math functions APIs guarantee both CUDA and CPU functionality, making it more straightforward to write `__host__ __device__` functions without being concerned whether the underlying intrinsics will build and work.
```

Not sure if compatibility was the best word, maybe this would read a bit better? Don't feel too strong either way

Reply:
`compatibility` here is kind of intentional because the purpose of these functions is to unify the cpu and gpu intrinsics to make sure we can compile them even if CUDA is not available.

Reply:
(therefore I used the term compatible to mean that the compilation would not fail).